pkg/operator/controller/controller_dns_daemonset.go

```diff
@@ -107,13 +110,56 @@ func desiredDNSDaemonSet(dns *operatorv1.DNS, coreDNSImage, kubeRBACProxyImage s
 		switch c.Name {
 		case "dns":
 			daemonset.Spec.Template.Spec.Containers[i].Image = coreDNSImage
+			if dns.Spec.UpstreamResolvers.TransportConfig.TLS != nil && dns.Spec.UpstreamResolvers.TransportConfig.TLS.CABundle.Name != "" {
+				vol, volMount := caBundleCMVolAndVolMount(dns.Spec.UpstreamResolvers.TransportConfig.TLS.CABundle.Name, dns.Spec.UpstreamResolvers.TransportConfig.TLS.ServerName, caBundleRevisionMap)
+				daemonset.Spec.Template.Spec.Volumes = append(daemonset.Spec.Template.Spec.Volumes, *vol)
+				daemonset.Spec.Template.Spec.Containers[i].VolumeMounts = append(daemonset.Spec.Template.Spec.Containers[i].VolumeMounts, *volMount)
+			}
+			for _, server := range dns.Spec.Servers {
+				if server.ForwardPlugin.TransportConfig.TLS != nil && server.ForwardPlugin.TransportConfig.TLS.CABundle.Name != "" {
+					vol, volMount := caBundleCMVolAndVolMount(server.ForwardPlugin.TransportConfig.TLS.CABundle.Name, server.ForwardPlugin.TransportConfig.TLS.ServerName, caBundleRevisionMap)
+					daemonset.Spec.Template.Spec.Volumes = append(daemonset.Spec.Template.Spec.Volumes, *vol)
+					daemonset.Spec.Template.Spec.Containers[i].VolumeMounts = append(daemonset.Spec.Template.Spec.Containers[i].VolumeMounts, *volMount)
+				}
+			}
 		case "kube-rbac-proxy":
 			daemonset.Spec.Template.Spec.Containers[i].Image = kubeRBACProxyImage
 		}
 	}
 	return daemonset, nil
 }
 
+// caBundleCMVolAndVolMount takes a CA bundle ConfigMap name and a TLS server name, and returns
+// the ConfigMap Volume and VolumeMount to be used for the DaemonSet.
+func caBundleCMVolAndVolMount(caBundleName string, serverName string, caBundleRevisionMap map[string]string) (*corev1.Volume, *corev1.VolumeMount) {
+	caBundleConfigmapName := CABundleConfigMapName(caBundleName)
+	caBundleVolumeName := caBundleConfigmapName.Name
+	caBundleFilename := "ca-bundle.crt"
```

Comment:
Should this be in names.go? I notice it is reference in code (this, testing, template) about 4 times.